seqr/views/utils/file_utils.py

```diff
@@ -86,25 +94,25 @@ def save_uploaded_file(request, process_records=None):
         raise ValueError("Received %s files instead of 1" % len(request.FILES))
 
     # parse file
-    stream = request.FILES.values()[0]
+    stream = list(request.FILES.values())[0]
     filename = stream._name
 
     json_records = parse_file(filename, stream)
     if process_records:
         json_records = process_records(json_records, filename=filename)
 
     # save json to temporary file
-    uploaded_file_id = hashlib.md5(str(json_records)).hexdigest()
+    uploaded_file_id = hashlib.md5(str(json_records).encode('utf-8')).hexdigest()
     serialized_file_path = _compute_serialized_file_path(uploaded_file_id)
-    with gzip.open(serialized_file_path, "w") as f:
+    with gzip.open(serialized_file_path, "wt") as f:
         json.dump(json_records, f)
 
     return uploaded_file_id, filename, json_records
 
 
 def load_uploaded_file(upload_file_id):
     serialized_file_path = _compute_serialized_file_path(upload_file_id)
-    with gzip.open(serialized_file_path) as f:
+    with gzip.open(serialized_file_path, "rt") as f:
```

Comment:
`t` is the default so why change it?

Reply:
The open modes of gzip are:
```
binary mode: 'r', 'rb', 'a', 'ab', 'w', 'wb', 'x', 'xb' 
text mode : 'rt', 'at', 'wt', or 'xt'
```
So, 'r' is a binary mode.

Reply:
Where are you seeing that? In the documentation I see the `b` flag specifies binary, the `t` flag specifies text, and text is the default so if neither is provided it is text mode: https://docs.python.org/3/library/functions.html#open

Reply:
The document you are looking at is for `open`. But in this case, it is about `gzip.open`.